app/src/main/java/com/github/geohunt/app/database/DatabasePathReference.java

```diff
@@ -0,0 +1,61 @@
+package com.github.geohunt.app.database;
+
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * This interface defines a reference to an abstract NoSQL Database, it can be
+ * used to read, write to changes in data at that location.
+ */
+public interface DatabasePathReference {
+    /**
+     * Returns a CompletableFuture object that represents a pending operations to retrieve
+     * a single object from the Database location referred to by this instance of DatabasePathReference
+     * @param classType the type of the class to be returned
+     * @return a CompletableFuture representing a pending operation to retrieve a single
+     * object from the Database

```

Comment:
This field duplicates the description that is given 2 lines above, one or the other should be removed in my opinion.

Reply:
I honestly don't think that it's so much of an issue here. In my opinion, this description can stay there because it will be easier in the future to just see the description of a function in the IDE.